gpsd_format/drivers.py

```diff
@@ -276,58 +272,55 @@ def close(self):
     def closed(self):
         return self._f.closed
 
-
-class GZIP(BaseDriver):
-
-    name = 'gzip'
-    extensions = 'gz',
-    modes = ('r', 'w', 'a')
-    compression = True
-
-    def __init__(self, f, mode='r', **kwargs):
-
-        def reader(f, **kwargs):
-            return gzip.GzipFile(fileobj=f, mode=mode, **kwargs)
-
-        def writer(f, **kwargs):
-            return gzip.GzipFile(fileobj=f, mode=mode, **kwargs)
-
-        BaseDriver.__init__(
-            self,
-            f=f, mode=mode,
-            reader=reader,
-            writer=writer,
-            modes=self.modes,
-            name=self.name,
-            **kwargs
-        )
-
-    def close(self):
-        self.obj.close()
-        self._f.close()
-
-
-# class BZ2(BaseDriver):
-#
-#     name = 'bz2'
-#     extensions = 'bz2',
-#     modes = ('r', 'w', 'a')
-#     compression = True
-#
-#     def __init__(self, f, mode='r', **kwargs):
-#
-#         BaseDriver.__init__(
-#             self,
-#             f=f, mode=mode,
-#             reader=bz2.BZ2File,
-#             writer=bz2.BZ2File,
-#             modes=self.modes,
-#             name=self.name,
-#             **kwargs
-#         )
-#
-#
-# class TAR(BaseDriver):
+try:
+    import gzip
+except:
+    pass
+else:
+    class GZIP(FileDriver):
+        name = 'gzip'
+        extensions = 'gz',
+        modes = ('r', 'w', 'a')
+        compression = True
+
+        def reader(self, f, *args, **kwargs):
+            return gzip.GzipFile(fileobj=f, *args, **kwargs)
+
+        def writer(self, f, *args, **kwargs):
+            return gzip.GzipFile(fileobj=f, *args, **kwargs)
+
+try:
+    import lzma
```

Comment:
I thought about the best way to handle registering drivers with non-critical dependencies and I think the solution is to just let an exception be raised when `gpsdio.open()` attempts to open a file.  It would be confusing if documented components just mysteriously aren't available in some situations.  I'm ok with using this workaround to unblock some critical tickets for now but I don't think its the best solution long-term.